pkg/ipam/crd.go

```diff
@@ -585,6 +616,9 @@ func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Famil
 			if n.isIPInReleaseHandshake(ip) {
 				continue // IP not available
 			}
+			if ipInfo.Owner != "" {
```

Comment:
I guess this is needed for proper function of the mechanism of https://github.com/cilium/cilium/pull/21379? Since I think the feature needs more design work (see other comment) it might be better to revert the previous PR (or only apply this change here to ensure it's at least working correctly).

Reply:
ok, I can commit this merge first to make sure the previous mr feature is complete

Reply:
#21715

Reply:
Thanks!